lib/Backend/BackwardPass.cpp

```diff
@@ -2107,6 +2107,16 @@ BackwardPass::DeadStoreTypeCheckBailOut(IR::Instr * instr)
         return;
     }
 
+    // If oldBailoutKind is equivTypeCheck and appropriate opnd doesn't have initial/final type
+    // then leave alone the bailout
+    if (oldBailOutKind == IR::BailOutFailedEquivalentTypeCheck || oldBailOutKind == IR::BailOutFailedEquivalentFixedFieldTypeCheck)
+    {
+        if (!(propertySymOpnd->HasInitialType() || propertySymOpnd->HasFinalType()))
```

Comment:
If that's the case, then can you find out how we got here with an equiv bailout kind?